genai-perf/tests/test_llm_profile_data_parser.py

```diff
@@ -740,3 +935,147 @@ def test_empty_response(self, mock_read_write: pytest.MonkeyPatch) -> None:
             },
         ],
     }
+
+    tensorrtllm_engine_profile_data = {
```

Comment:
I think it would be cleaner to define this in the function its derivative is going to be used in. There should be a way to invoke the monkey patcher after some test case code has already been run (e.g. this definition) perhaps using .context():

https://docs.pytest.org/en/latest/reference/reference.html#pytest.MonkeyPatch.context